### PR TITLE
new function to apply LSF

### PR DESCRIPTION
**Pull request by ajwheeler**
This introduces a function that calculates the matrix which maps spectra to LSF-convolved spectra, assuming the LSF is a gaussian with width set by a constant R = lambda/Delta lambda.  It assumes that the wavelength grid is locally-linear and dense, or it will not generally be normalized correctly.  In the future, I plan to automate the process of synthesizing on a dense grid, convolving with the LSF, and downsampling automatic, since it involves a few steps.

Here's the Ca triplet (what the Gaia RVS uses), with a realistic-ish LSF imposed.
image

### CODE REVIEW

**Comment by ajwheeler** [comment]
I just realized that this is totally untenable for large spectra.  I will address by not explicitly constructing a matrix.

**Comment by ajwheeler** [comment]
This now computes the convolved spectrum directly.  It's not as fast as would be idea.  (It takes about 2 mins to on the full visible with a dense wl grid, but we can look into efficient methods later.  I'm not really sure what the options are since the convolution kernel in wl-dependent.  Maybe we could transform to a basis where it's constant?  Then the normalization might be tricky...

**Comment by mabruzzo** [comment]
What exactly is a LSF-convolved spectra? Does this accounting for the fact that an instrument has finite resolution?

**Comment by ajwheeler** [comment]
This doesn’t account for the pixels on the detector, but it account for the line spread from the instrument optics, which generally has this functional form to first (second?) order. 

Dealing with the detector pixels is also important, but we aren’t doing this at the moment.

File: src/LSF.jl [hunk 1]
```diff
@@ -0,0 +1,22 @@
+normal_pdf(Δ, σ) = exp(-0.5*Δ^2 / σ^2) / √(2π) / σ
+
+"""
+Returns the matrix which transforms a spectrum to its LSF-convolved version for spectral resolution
```
Comment (mabruzzo) [inline src/LSF.jl:4]: could you maybe just spell out "line spread function"? Other than that, this seems great

File: src/LSF.jl [hunk 2]
```diff
@@ -1,11 +1,12 @@
 normal_pdf(Δ, σ) = exp(-0.5*Δ^2 / σ^2) / √(2π) / σ
 
 """
-Returns the matrix which transforms a spectrum to its LSF-convolved version for spectral resolution 
+Returns the matrix which a gaussian line spread function with constant spectral resolution,
```
Comment (mabruzzo) [inline src/LSF.jl:4]: I don't mean to nitpick, but there's a typo here

Comment (ajwheeler) [reply src/LSF.jl]: Not only that, it was factually wrong!  thanks.

**Pull request merged**